### ISSUE TITLE
Null-Propagation Quick-Fix Suggests Invalid Fix

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#16114]
**Version Used**: d15prerel/26022.07

Simple example:

```C#
namespace ConsoleApp1
{
    class Base
    {
        public Base(int i) { }
    }

    class Program : Base
    {
        public Program(string a) :
            base(a != null ? a.Length : throw new ArgumentNullException(nameof(a)))
        {
        }
    }
}
```

The IDE suggests moving `base(a != null ? a.Length : throw new ArgumentNullException(nameof(a)))` to use null propagation as follows: `base(a?.Length)`. This changes behavior, removing the ArgumentNullException, and it also no longer compiles, as `a?.Length` is an `int?`, not an `int`.

/cc @Pilchie @dpoeschl

**Comment by alrz** [comment]
It should probably be  `a?.Length ?? throw new ArgumentNullException(nameof(a))`.

**Comment by 333fred** [comment]
No, that also changes behavior. Length isn't going to return null, but you can easily imagine a different property that returns null, even if the original object isn't null.

**Comment by alrz** [comment]
@333fred When you use `??` the nullable type will be unwrapped.

**Comment by 333fred** [comment]
@alrz that doesn't stop the change of behavior. If the original object isn't null, but the property is, `a?.Length ?? throw new ArgumentNullException(nameof(a))` will throw, but `a != null ? a.Length : throw new ArgumentNullException(nameof(a))` won't. You could use `(a ?? throw new ArgumentNullException(nameof(a))).Length`, but that looks terrible to me and is hard to parse visually.

**Comment by alrz** [comment]
@333fred Right, if the property is a reference type, that's true. I'm just saying that in your example it doesn't change the behavior so it will be safe to do it in this specific case.

**Comment by davkean** [comment]
I believe @CyrusNajmabadi recently fixed this, I can't find the PR though. While it still repro's in 26028.1, it doesn't repro in my RoslynDev hive with recent bits from master.

**Comment by Pilchie** [comment]
@CyrusNajmabadi Can you confirm a fix here?

**Comment by CyrusNajmabadi** [comment]
Yes.  This has been fixed.  We were  missing the check that the code down the when-true/false path was 'null' as appropriate.

**Comment by alrz** [comment]
@CyrusNajmabadi In case `a.Length` is a value type it could use ?? operator. Should that be another fix then or we don't usually consider types to suggest fixes?

**Comment by CyrusNajmabadi** [comment]
Seems like a reasonable thing a new feature could provide.
